Accelerator unit tests are failing due to `TypeError: load_model() got an unexpected keyword argument 'device'`

Several unit tests are failing with the error messages similar to `FAILED tests/test_accelerator.py::AcceleratorTester::test_save_load_model_use_safetensors - TypeError: load_model() got an unexpected keyword argument 'device'` which appears to originate from https://github.com/huggingface/accelerate/blob/main/src/accelerate/checkpointing.py#L208

For the versions of the packages below, running `make test` or `pytest tests/test_accelerator.py` on the main branch fails on my machine:

```
$ conda list | grep torch
ffmpeg                    4.3                  hf484d3e_0    pytorch
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
pytorch                   2.3.1           py3.11_cuda12.1_cudnn8.9.2_0    pytorch
pytorch-cuda              12.1                 ha16c6d3_5    pytorch
pytorch-mutex             1.0                        cuda    pytorch
torchaudio                2.3.1               py311_cu121    pytorch
torchdata                 0.7.1.dev20240703+cpu          pypi_0    pypi
torchtriton               2.3.1                     py311    pytorch
torchvision               0.18.1              py311_cu121    pytorch
```

--- a/setup.py
+++ b/setup.py
@@ -76,7 +76,7 @@ setup(
         "pyyaml",
         "torch>=1.10.0",
         "huggingface_hub>=0.21.0",
-        "safetensors>=0.3.1",
+        "safetensors>=0.4.3",
     ],
     extras_require=extras,
     classifiers=[

